legwork/source.py

```diff
@@ -92,18 +114,43 @@ class Source():
     ------
     ValueError
         If both ``f_orb`` and ``a`` are missing.
+        If only part of the position, inclination, and polarization are supplied.
         If array-like parameters don't have the same length.
 
     AssertionError
         If a parameter is missing units
     """
+
     def __init__(self, m_1, m_2, ecc, dist, n_proc=1, f_orb=None, a=None,
+                 theta=None, phi=None, ra=None, dec=None, psi=None, inc=None,
                  gw_lum_tol=0.05, stat_tol=1e-2, interpolate_g=True,
                  interpolate_sc=True, sc_params={}):
         # ensure that either a frequency or semi-major axis is supplied
         if f_orb is None and a is None:
             raise ValueError("Either `f_orb` or `a` must be specified")
 
+        # ensure that if and of psi, inc, and position are supplied that
+        # the other parameters are also supplied
+        if inc is not None and ((theta is None or phi is None) and
+                                (ra is None or dec is None)):
+            raise ValueError("If you specify the inclination, you must also specify a sky position.")
+
+        if psi is not None and ((theta is None or phi is None) and
+                                (ra is None or dec is None)):
+            raise ValueError("If you specify the polarization, you must also specify a sky position.")
+
+        if (theta is not None and phi is not None) or (ra is not None and dec is not None):
+            if inc is None:
+                print("assume a randomly inclined source")
```

Comment:
This sort of implies that it is singular so could be interpreted as a single random value. I think you should make it seem plural

Reply:
Changed.